### ISSUE TITLE
s390x support for Etcd

### ISSUE THREAD

**Issue by Nayana-ibm** [coreos/etcd#9409]
Hi, 

We would like to know if Etcd CI can be extended to support z system. We are ready to provide required support from IBM.

We have gone through https://github.com/coreos/etcd/blob/master/Documentation/op-guide/supported-platform.md#supporting-a-new-platform and have an understanding of requirements.

1. An "official" maintainer for the platform with clear motivation; someone must be responsible for taking care of the platform. 
--> IBM z team will maintain the CI
2.  Set up CI for build; etcd must compile.
--> Travis and SemaphoreCI are are not supproted by IBM z systems. We can provide Jenkins slave agent under current etcd Jenkins master (https://jenkins-etcd-public.prod.coreos.systems/)
3.  Set up CI for running unit tests; etcd must pass simple tests.
--> We will ensure to run etcd unit tests in the Jenkins slave node for z systems
4.  Set up CI (TravisCI, SemaphoreCI or Jenkins) for running integration tests; etcd must pass intensive tests.
-->We will try to implement Jenkins tests similar to what has been implemented for PPC platform.
5.  (Optional) Set up a functional testing cluster; an etcd cluster should survive stress testing

Please let us know your view/comments.

**Comment by gyuho** [comment]
Sounds good with test plan. Do we also need release container image for s930x?

**Comment by Nayana-ibm** [comment]
@gyuho Thanks for inputs.
We are referring to #6818 where Etcd CI is extended for ppc64. On similar lines we want to add support for s390x. 
As mentioned in #6818, jenkins slave node can be added to CI  with setting authorized_keys.  is it  possible to add slave node (s390x vm) using JNLP?

Also, could you please let us know know what hardware configuration is required? Which distribution to prefer?

**Comment by Nayana-ibm** [comment]
@gyuho Could you please provide inputs on this?

**Comment by gyuho** [comment]
@Nayana-ibm Can you email gylee@redhat.com? We need set up Jenkins job with your remote machine.

**Comment by Nayana-ibm** [comment]
@gyuho  If I understand correctly, you need access to the s390x VM to setup Jenkins slave agent. If yes, as part of IBM's policy, we need external (non-IBM) developers to acknowledge agreement to access the system z VM. Let us know if that is OK with you. If not, we can setup the node for you and you add it in etcd Jenkins server.
Also, once Jenkins is ready, will have s390x binaries available? will be published on [Etcd GitHub](https://github.com/coreos/etcd/releases)?

**Comment by gyuho** [comment]
@Nayana-ibm Yeah that's why I asked to email the IP address of VM. For s390x release, we can include s390x docker images, but not compiled binaries.

**Comment by Nayana-ibm** [comment]
@gyuho Could you please let us know the VM configuration (CPU, Memory, Storage) ? Also, which distribution (Ubuntu/RHEL) is preferable?  Once we get this info , will arrange and provide the VM to you. 
Also, Ts & Cs needs to be signed. You will be signing it? or someone else from community?

**Comment by gyuho** [comment]
2 vCPUs, 10 GB memory, 50 GB SSD should be enough.

> Also, which distribution (Ubuntu/RHEL) is preferable?

Let's use RHEL.

> Ts & Cs needs to be signed.

What is this?

**Comment by Nayana-ibm** [comment]
@gyuho I have sent an mail to gylee@redhat.com with machine details.

**Comment by Nayana-ibm** [comment]
@gyuho I have sent an email to you.Please check.

**Comment by Nayana-ibm** [comment]
@gyuho Are you working on new Jenkins setup for Etcd? Or Any other dev is looking into this?

**Comment by hexfusion** [comment]
@Nayana-ibm I have some information, would you mind sending me an email sam@hexfusion.com and I can follow up, thanks.

**Comment by xnox** [comment]
@Nayana-ibm @gyuho I see that there are no s390x release binaries for most recent releases =(

Is there anything I can help with? Do we still need jenkins? Travis has s390x support now, could we use that for CI too?

**Comment by nirmannarang** [comment]
@xnox s390x cross-build support has been added to the master branch, changelog updated in [CHANGELOG-3.5.md](https://github.com/etcd-io/etcd/blob/master/CHANGELOG-3.5.md)
Expecting s390x binaries to be released in etcd v3.5.x.
